lib/puppet/configurer.rb

```diff
@@ -356,7 +356,21 @@ def run_internal(options)
         )
       end
 
+      session = Puppet.lookup(:http_session)
+      fs = session.route_to(:fileserver)
+
+      temp_value = options[:pluginsync]
+      begin
+        response = fs.get_file_metadata(path: '/plugins', environment: @environment)
```

Comment:
Did not find a better way to find out if a environment exists on the server

```
fs.get_file_metadata(path:"/plugins", environment:"dev")
[#<Puppet::HTTP::Response:0x000055d998775728 @nethttp=#<Net::HTTPOK 200 OK readbody=true>, @url=#<URI::HTTPS https://inverse-ton.delivery.puppetlabs.net:8140/puppet/v3/file_metadata/plugins?links=manage&checksum_type=md5&source_permissions=ignore&environment=dev>>, #<Puppet::FileServing::Metadata:0x000055d9986b50e0 @owner=993, @group=990, @mode=420, @checksum_type="ctime", @checksum="{ctime}2021-07-19 10:51:56 UTC", @ftype="directory", @destination=nil, @source=nil, @content_uri=nil, @path="/opt/puppetlabs/puppet/modules/pe_infrastructure/lib", @links=:manage>]
```

Tried different paths for `get_file_metadata` but only `/plugins` worked
```
fs.get_file_metadata(path:"/pluginfacts", environment:"dev")
Not Found
fs.get_file_metadata(path:"/plugins/inifile/lib", environment:"dev")
Not Found
fs.get_file_metadata(path:"/plugins/pe_infrastructure/lib", environment:"dev")
Not Found
fs.get_file_metadata(path:"/plugins/modules/pe_infrastructure/lib", environment:"dev")
Not Found

```
On server logs
```
2021-07-20T14:41:01.443Z INFO  [qtp1973295078-42] [puppetserver] Puppet Not Found: Could not find file_metadata pluginfacts
2021-07-20T14:42:23.817Z INFO  [qtp1973295078-44] [puppetserver] Puppet Not Found: Could not find file_metadata plugins/inifile/lib
2021-07-20T14:43:41.745Z INFO  [qtp1973295078-44] [puppetserver] Puppet Not Found: Could not find file_metadata plugins/pe_infrastructure/lib
2021-07-20T14:43:47.603Z INFO  [qtp1973295078-38] [puppetserver] Puppet Not Found: Could not find file_metadata plugins/modules/pe_infrastructure/lib
```

Reply:
Using `/plugins` makes sense, though in theory someone could override `Puppet[:pluginsource]`. So to make sure we're using the same value, you could extract the HTTP request path as:

```
irb(main):003:0> URI(Puppet[:pluginsource]).path
=> "/plugins"
```

And pass that to `get_file_metadata`